rust/serving/src/app/jetstream_proxy.rs

```diff
@@ -71,27 +67,28 @@ pub(crate) async fn jetstream_proxy<T: Clone + Send + Sync + Store + 'static>(
 }
 
 async fn sync_publish_serve<T: Send + Sync + Clone + Store>(
-    State(mut proxy_state): State<ProxyState<T>>,
+    State(proxy_state): State<Arc<ProxyState<T>>>,
     headers: HeaderMap,
     body: Bytes,
 ) -> impl IntoResponse {
-    let id = extract_id_from_headers(&headers);
+    let id = extract_id_from_headers(&proxy_state.tid_header, &headers);
 
     // Register the ID in the callback proxy state
-    let notify = proxy_state.callback.register(id.clone());
+    let notify = proxy_state.callback.clone().register(id.clone());
 
     if let Err(e) = publish_to_jetstream(
-        proxy_state.stream,
+        proxy_state.stream.clone(),
```

Comment:
is this clone too much for every request? we know that `String` is a `const` since it is stream name.

Reply:
Currently, a lot of clones are happening for Router states. I will work on a separate PR to avoid those. Need to figure out whether to use `Arc` of the struct or `Arc` of individual fields, some handlers only need part of the application state. Since config lives for the duration of the application, we could make all strings to `&'static str` with `Box::leak` to avoid all cloning or use `Arc<str>` instead of `String` for cheaper cloning.

Reply:
please do as a follow up

Reply:
https://github.com/numaproj/numaflow/issues/2296